packages/flutter/lib/src/widgets/mimic.dart

```diff
@@ -19,6 +23,73 @@ class MimicableKey {
   }
 }
 
+class MimicOverlayEntry {
+  MimicOverlayEntry._(this._key) {
+    _overlayEntry = new OverlayEntry(builder: _build);
+  }
+
+  MimicableKey _key;
+  OverlayEntry _overlayEntry;
+
+  // Animation state
+  GlobalKey _targetKey;
+  Curve _curve;
+  Performance _performance;
+
+  Future animateTo({
+    GlobalKey targetKey,
+    Duration duration,
+    Curve curve: Curves.linear
+  }) {
+    assert(_key != null);
+    assert(_overlayEntry != null);
+    assert(targetKey != null);
+    assert(duration != null);
+    assert(curve != null);
+    _targetKey = targetKey;
+    _curve = curve;
+    // TODO(abarth): Support changing the animation target when in flight.
+    assert(_performance == null);
+    _performance = new Performance(duration: duration)
+      ..addListener(_overlayEntry.markNeedsBuild);
+    return _performance.play();
+  }
+
+  void stopMimic() {
+    _targetKey = null;
+    _curve = null;
+    _performance?.stop();
+    _performance = null;
+    _key.stopMimic();
+    _key = null;
+    _overlayEntry.remove();
+    _overlayEntry = null;
+  }
+
+  Widget _build(BuildContext context) {
+    assert(_key != null);
+    assert(_overlayEntry != null);
+    Rect bounds = _key.globalBounds;
+    if (_targetKey != null) {
+      assert(_performance != null);
+      assert(_curve != null);
+      RenderBox box = _targetKey.currentContext?.findRenderObject();
+      if (box != null) {
+        // TODO(abarth): Handle non-translation coordinate changes.
```

Comment:
should probably assert that the transform between the two really is just a translation. Assuming we're the overlay and the target guy is referenceBox, this is something like:

```dart
 assert(() {
    // find the chain of renderers from us to the referenceBox                                                                                                                                                          
    List<RenderBox> descendants = <RenderBox>[];
    RenderBox node = referenceBox;
    while (node != renderer) {
      descendants.add(node);
      node = node.parent;
      assert(node != null);
    }
    // determine the transform that gets our coordinate system to be like theirs                                                                                                                                                  
    Matrix4 transform = new Matrix4.identity();
    for (RenderBox descendant in descendants.reversed)
      descendant.applyPaintTransform(transform);
    return MatrixUtils.getAsTranslation(transform) != null;
 });
```